addons/account/static/src/js/bills_upload.js

```diff
@@ -1,160 +1,170 @@
-            Controller: KanbanController.extend({_dropZone: '.o_content'}, UploadDocumentMixin, {
-                buttons_template: 'DocumentsKanbanView.buttons',
-                events: Object.assign({}, KanbanController.prototype.events, UploadDocumentMixin.events),
-            }),
-        }),
-    });
-
-    var DashboardKanbanController = KanbanController.extend({_dropZone: '.o_kanban_record'}, UploadDocumentMixin, {
-        events: Object.assign({}, KanbanController.prototype.events, UploadDocumentMixin.events),
-
-        _setJournalContext: function(ev) {
-            var kanbanRecord = $(ev.currentTarget).closest('.o_kanban_record').data('record');
-            this.initialState.context['default_journal_id'] = kanbanRecord.id;
-            let journal_type = kanbanRecord.state.data.type;
-            if (journal_type == 'sale') {
-                this.initialState.context['default_move_type'] = 'out_invoice';
-            } else if (journal_type == 'purchase') {
-                this.initialState.context['default_move_type'] = 'in_invoice';
-            }
-        },
-
-        _onUpload: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onUpload.apply(this, arguments);
-        },
-
-        _onDrop: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onDrop.apply(this, arguments);
-        },
-    });
-
-    var DashboardKanbanView = KanbanView.extend({
-        config: Object.assign({}, KanbanView.prototype.config, {
-            Controller: DashboardKanbanController,
-        }),
-    });
-
-    viewRegistry.add('account_tree', DocumentsListView);
-    viewRegistry.add('account_documents_kanban', DocumentsKanbanView);
-    viewRegistry.add('account_dashboard_kanban', DashboardKanbanView);
-});
+    async onFileUploaded(file) {
+        let att_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        }
+        const att_id = await this.orm.call("ir.attachment", "create", [att_data], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process.push(att_id);
+    }
+
+    async onUploadComplete() {
+        const action = await this.orm.call("account.journal", "create_document_from_attachment", ["", this.attachment_ids_to_process], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process = [];
+        this.action.doAction(action);
+    }
+}
+AccountFileUploader.components = {
+    FileUploader,
+}
+AccountFileUploader.template = "account.HiddenFileUploader";
+
+class AccountDropZone extends Component {}
+AccountDropZone.defaultProps = {
+    visibilityClass: "d-none",
+    dragOver: () => {},
+    dragLeave: () => {},
+    drop: () => {},
+}
+AccountDropZone.template = "account.DropZone";
+
+const AccountMoveDropzoneFunctions = {
```

Comment:
This is a mixin. Mixins have issues with regards to composability and they clobber the namespace. A hook can solve the same problems without these drawbacks. Come find me if it's not obvious to you how to convert this to a hook. It also looks like a good bit of this logic could be moved to the dropzone component itself, which would also allow you to use a ref instead of querying the DOM lower in this file.